Providers/SelectedListManager.tscn

```diff
@@ -0,0 +1,160 @@
+
+
+# Function to apply values to selected enemy
+func apply_values_to_enemy(enemy_data: Enemy, parameters: Dictionary):
+	for parameter_name in parameters.keys():
+		var parameter_value = parameters[parameter_name]
+		match parameter_name:
+			\"named_enemy_params_id\":
+				enemy_data.named_enemy_params_id = parameter_value
+			\"raid_boss_id\":
+				enemy_data.raid_boss_id = parameter_value
+			\"scale\":
+				enemy_data.scale = parameter_value
+			\"lv\":
+				enemy_data.lv = parameter_value
+			\"hm_preset_no\":
+				enemy_data.hm_preset_no = parameter_value
+			\"start_think_tbl_no\":
+				enemy_data.start_think_tbl_no = parameter_value
+			\"repop_num\":
+				enemy_data.repop_num = parameter_value
+			\"repop_count\":
+				enemy_data.repop_count = parameter_value
+			\"enemy_target_types_id\":
+				enemy_data.enemy_target_types_id = parameter_value
+			\"montage_fix_no\":
+				enemy_data.montage_fix_no = parameter_value
+			\"set_type\":
+				enemy_data.set_type = parameter_value
+			\"infection_type\":
+				enemy_data.infection_type = parameter_value
+			\"time_type\":
+				enemy_data.time_type = parameter_value
+			\"is_boss_gauge\":
+				enemy_data.is_boss_gauge = parameter_value
+			\"is_boss_bgm\":
+				enemy_data.is_boss_bgm = parameter_value
+			\"is_manual_set\":
+				enemy_data.is_manual_set = parameter_value
+			\"is_area_boss\":
+				enemy_data.is_area_boss = parameter_value
+			\"is_blood_enemy\":
+				enemy_data.is_blood_enemy = parameter_value
+			\"is_highorb_enemy\":
+				enemy_data.is_highorb_enemy = parameter_value
+			\"blood_orbs\":
+				enemy_data.blood_orbs = parameter_value
+			\"high_orbs\":
+				enemy_data.high_orbs = parameter_value
+			\"experience\":
+				enemy_data.experience = parameter_value
+			\"drops_table\":
+				enemy_data.drops_table.id = parameter_value
+			# Add more cases for other parameters as needed
+			_:
+				# Handle unknown parameter names
+				print(\"Unknown parameter name:\", parameter_name)
+
+# Function to check the list for type and removes if it exists already
+func _check_list_for_type(placemark: GenericPlacemark, data: Object):
+	var entry = {\"placemark\": placemark, \"data\": data}
+
+	var index = _find_entry_index(placemark)
+	if index != -1:
+		var panel = placemark.get_node(\"SelectionPanel\")
+		if panel:
+			panel.visible = false
+			selected_list.remove(index)
+			reload_details_panel()
+	else:
+		selected_list.append(entry)
+		detailspanel.show_details_of(data)
+		highlight()
+	if selected_list.size() > 1:
+		set_display_name()
+		
+func set_display_name():
+	print(\"Selected Nodes: \" + str(selected_list.size())) 
+
+# Function to find the index of an entry in the list
+func _find_entry_index(placemark: GenericPlacemark) -> int:
+	for i in range(selected_list.size()):
+		if selected_list[i][\"placemark\"] == placemark:
+			return i
+	return -1
+
+# Adds the highlight selection around anything inside the selected_list
+func highlight():
```

Comment:
SelectedListManager is a "global" kind of deal, I think it's cleaner if it doesn't care about the UI elements that sit under it, and instead is just used as a "reference" or the source of truth for what's highlighted and what isn't, and have each individual UI element worry about their own style, instead of having SelectedListManager have to handle their style